services/SocialShareService.php

```diff
@@ -0,0 +1,224 @@
+        $encodedDescription = urlencode($description);
+
+        if ($platform === self::PLATFORM_BLUESKY) {
+            $encodedText = urlencode($description . ' ' . $permalink);
+            $url = str_replace(
+                ['{url}', '{text}'], 
+                ['', $encodedText], 
+                $this->platforms[$platform]['urlPattern']
+            );
+        } else {
+            $url = str_replace(
+                ['{url}', '{text}'], 
+                [$encodedPermalink, $encodedDescription], 
+                $this->platforms[$platform]['urlPattern']
+            );
+        }
+
+        if (!empty($additionalParams)) {
+            $url .= '&' . http_build_query($additionalParams);
+        }
+
+        return $url;
+    }
+
+    /**
+     * Generate share link HTML for a specific platform
+     * 
+     * @param string $platform Platform identifier
+     * @param string $permalink Permalink to the content
+     * @param string $description Content description
+     * @param array $options Rendering options
+     * @return string HTML link tag
+     */
+    public function createShareLink($platform, $permalink, $description, $options = [])
+    {
+        if (!isset($this->platforms[$platform])) {
+            throw new \InvalidArgumentException("Unsupported platform: {$platform}");
+        }
+
+        $options = array_merge($this->defaultOptions, $options);
+
+        $url = $this->getSharingUrl($platform, $permalink, $description);
+        $icon = $this->getPlatformIcon($platform);
+
+        return Html::a($icon, $url, $options);
+    }
+
+    /**
+     * Generate HTML for all supported share links
+     * 
+     * @param mixed $object Content object that provides getContentDescription()
+     * @param string $permalink Permalink to the content
+     * @param array $options Rendering options
+     * @return string HTML with all share links
+     */
+    public function renderShareLinks($object, $permalink, $options = [])
+    {
+        $description = method_exists($object, 'getContentDescription') 
+            ? $object->getContentDescription() 
+            : '';
+
+        $links = [];
+
+        foreach (array_keys($this->platforms) as $platform) {
+            $links[] = $this->createShareLink($platform, $permalink, $description, $options);
+        }
+
+        return implode("\n", $links);
+    }
+
+    /**
+     * Get HTML for the platform icon
+     * 
+     * @param string $platform Platform identifier
+     * @return string HTML icon
+     */
+    protected function getPlatformIcon($platform)
+    {
+        if (!isset($this->platforms[$platform])) {
+            throw new \InvalidArgumentException("Unsupported platform: {$platform}");
+        }
+
+        $iconClass = $this->platforms[$platform]['iconClass'];
+        $iconColor = $this->platforms[$platform]['iconColor'];
+
+        return '<i class="' . $iconClass . '" style="font-size:16px;color:' . $iconColor . '">&nbsp;</i>';
```

Comment:
It can be replaced with:

```php
return Icon::get($iconClass)->color($iconColor)->style('font-size:16px');
```

but need remove `fa fa-` from all `iconClass`, e.g. `'iconClass' => 'facebook',`.

Reply:
> It can be replaced with:
> 
> ```
> return Icon::get($iconClass)->color($iconColor)->style('font-size:16px');
> ```
> 
> but need remove `fa fa-` from all `iconClass`, e.g. `'iconClass' => 'facebook',`.

This would require me to also add `->right()` to keep the correct position.

Reply:
Done! https://github.com/humhub-contrib/socialshare/pull/12/commits/dc4fd1ce35ecb7928364eea7c797eec9f174783b